tree/tree/inc/TLeaf.h

```diff
@@ -69,9 +69,14 @@ class TLeaf : public TNamed {
    virtual void     FillBasket(TBuffer &b);
    virtual Int_t   *GenerateOffsetArray(Int_t base, Int_t events) { return GenerateOffsetArrayBase(base, events); }
    TBranch         *GetBranch() const { return fBranch; }
-   virtual TLeaf   *GetLeafCount() const { return fLeafCount; }
+   virtual TLeaf   *GetLeafCount() const { return fLeafCount; } ///< Return a pointer to the leaf that stores the number of elements stored in this leaf. Only meaningful if the leaf contains a variable-length array.
    virtual TLeaf   *GetLeafCounter(Int_t &countval) const;
    virtual Int_t    GetLen() const;
+   /// Return the fixed length of this leaf.
+   /// If the leaf stores a fixed-length array, this is the size of the array.
+   /// If the leaf stores a non-array or a variable-sized array, this method returns 1.
+   /// If the leaf stores an array with 2 or more dimensions, this method returns the total number of elements in the
+   /// dimensions with static length: for example for float[3][2][] it would return 6.
    virtual Int_t    GetLenStatic() const { return fLen; }
```

Comment:
See also the documentation for fLen
```
Int_t fLen; ///<  Number of fixed length elements
```
which 'compactly' says the same thing (but it does sound too terse).

Reply:
I don't think anyone except you could deduce those four lines of documentation I wrote from "number of fixed length elements" 😅

Reply:
I agree :)